src/SACA.chpl

```diff
@@ -0,0 +1,207 @@
+module SACA {
+  //In this module, different algorithms to construct suffix array are provided
+
+  // The Chapel version of suffix array construction algorithm using skew algorithm
+  // Rewrite the algorithm and codes in paper
+  // "Simple Linear Work Suffix Array Construction" by Juha Karkkainen and Peter Sanders (2003)
+  // Dec.7, 2020
+
+  inline proc leq(a1:int, a2:int, b1:int, b2:int) {
+    // lexicographic order
+    return(a1 < b1 || a1 == b1 && a2 <= b2);
+  } // for pairs
+
+  inline proc leq(a1:int, a2:int, a3:int, b1:int, b2:int, b3:int) {
+    // lexicographic order
+    return(a1 < b1 || a1 == b1 && leq(a2, a3, b2, b3));
+  } // for pairs
```

Comment:
Additional parenthesis should be added to the logic statements above for clarity.

Reply:
I've added parenthesis in both of the `leq` procs and renamed them for clarity